src/main/groovy/io/seqera/wave/service/blob/impl/BlobCacheServiceImpl.groovy

```diff
@@ -209,9 +235,32 @@ class BlobCacheServiceImpl implements BlobCacheService {
      * @return The HTTP URI from the cached layer blob is going to be downloaded
      */
     protected String blobDownloadUri(RoutePath route) {
-        StringUtils.pathConcat(blobConfig.baseUrl, route.targetPath)
+        createPresignedGetUrl(blobConfig.storageBucket, route.targetPath)
     }
 
+    /**
+     *  Create a pre-signed URL to download an object in a subsequent GET request.
+     *  @param s3 bucket name
+     *  @param key in the s3 bucket
+     *  @return pre signed URL
+     */
+    private String createPresignedGetUrl(String bucketName, String keyName) {
+            GetObjectRequest objectRequest = GetObjectRequest.builder()
```

Comment:
Intellij is reporting a type error at this line 

<img width="841" alt="Screenshot 2024-01-15 at 21 16 49" src="https://github.com/seqeralabs/wave/assets/816968/1294102c-f3c1-4eea-8e72-f97c194478db">

Reply:
fixed